### PR TITLE
Update method name rule 

### PR DESCRIPTION
**Pull request by tneer**
Update method name rule to allow method names to start with a single lower case letter.

### CODE REVIEW

File: checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml [hunk 1]
```diff
@@ -162,7 +162,7 @@
       <property name="allowThrowsTagsForSubclasses" value="true"/>
     </module>
     <module name="MethodName">
-      <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>
+      <property name="format" value="^[a-z][a-zA-Z0-9_]*$"/>
```
Comment (bradcupit) [inline checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml]: +1, but I'm not sure we should have the `_` in there.
Perhaps a topic for another day.

**Comment by vladt** [comment]
Do you have examples of method names that need to be changed after this rule is enforced?

I'm thinking that we should enforce rules only because they really make our code better, not just because we can... :)

**Comment by tneer** [comment]
> Do you have examples of method names that need to be changed after this rule is enforced?

This is more permissive than the previous rule, so it's not that there are some that will need to be changed after the rule change, but more that there are some methods that won't need to be changed to be in compliance.  For example, https://github.com/sonatype/insight-scanner-hashing/blob/4d5ba94cb4b80e8149148562be04262a928f81f0/insight-scanner-hashing/src/test/java/com/sonatype/insight/scan/hash/internal/asm/AbstractClassDigestTest.java#L38 will be allowed by this change.  It is in violation of the current rule.

**Comment by vladt** [comment]
> This is more permissive than the previous rule, so it's not that there are some that will need to be changed after the rule change, but more that there are some methods that won't need to be changed to be in compliance. For example, https://github.com/sonatype/insight-scanner-hashing/blob/4d5ba94cb4b80e8149148562be04262a928f81f0/insight-scanner-hashing/src/test/java/com/sonatype/insight/scan/hash/internal/asm/AbstractClassDigestTest.java#L38 will be allowed by this change. It is in violation of the current rule.

I understand. My question is: Do we really need this rule (before or after your change)? What problems does it prevent?

**Comment by tneer** [comment]
>  Do we really need this rule (before or after your change)? What problems does it prevent?

I think we do.  I think we care about method names.  We don't want method names that start with a capital letter.  We don't want Δ instead of 'delta'.  We don't want _privateMethod().

**Pull request merged**